### PR TITLE
the entity class (player//enemy parent class)

### PR DESCRIPTION
**Pull request by ivan-BRHM**
#68 
Adding the parent class for enemies and players. Features so far include:
* emotes
* weapons and attacks
* hp and death

### CODE REVIEW

File: my_sprites.py [hunk 1]
```diff
@@ -50,6 +50,169 @@ class Sound(Enum):
...
+
+        # amount of seconds before the sprite can update
+        self.pause_timer = 0
+
+    @property
+    def max_hp(self):
+        return self._max_hp
+
+    @property
+    def cur_hp(self):
```
Comment (tobiasbp) [inline my_sprites.py:117]: Maybe this should be renamed to _hp_?

File: my_sprites.py [hunk 2]
```diff
@@ -50,6 +50,169 @@ class Sound(Enum):
...
+        return self.cur_hp
+
+    @cur_hp.setter
+    def cur_hp(self, new_hp):
+        self.cur_hp = max(0, min(new_hp, self.max_hp))
+
+    @property
+    def emotes(self):
```
Comment (tobiasbp) [inline my_sprites.py:125]: I think this was for drawing/updating. If we do that in the class, then we should probably remove this property?

File: my_sprites.py [hunk 3]
```diff
@@ -417,163 +517,35 @@ def __init__(
...
+            self.facing_dir = Direction.RIGHT
             return
         elif key == self.key_up:
             self.up_pressed = True
-            self._direction = Direction.UP
+            self.facing_dir = Direction.UP
         elif key == self.key_down:
             self.down_pressed = True
-            self._direction = Direction.DOWN
+            self.facing_dir = Direction.DOWN
```
Comment (tobiasbp) [inline my_sprites.py:548]: Why do you want to rename them and make them non _"private"_ (removing _ prefix) ?

Comment (ivan-BRHM) [reply my_sprites.py:548]: so, just rename the variable to _facing_dir?
- not make it into a proberty

**Review by tobiasbp — APPROVED** [review]

**Review by tobiasbp — APPROVED** [review]

**Pull request merged**